### PR TITLE
feat: Add all common cargo commands and more options

### PR DESCRIPTION
**Pull request by grant0417**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Improves existing spec

**What is the new behavior (if this is a feature change)?**
Improves `cargo` support, adds all of the "common" cargo commands and adds most options for them. New commands: check, clean, doc, new, run, bench, install, uninstall, search

**Additional info:**
None

### CODE REVIEW

File: src/cargo.ts [hunk 1]
```diff
@@ -39,98 +39,356 @@ const binList: Fig.Generator = {
...
         {
-          name: "--bins",
-          description: "Build all binaries",
+          name: "--all",
+          description: "Alias for workspace (deprecated)",
+          hidden: true,
+        },
+        {
+          name: "--workspace",
+          description: "Build all packages in workspace",
```
Comment (Milo123459) [inline src/cargo.ts:111]:
```suggestion
          description: "Build all packages in the workspace",
```

File: src/cargo.ts [hunk 2]
```diff
@@ -39,98 +39,356 @@ const binList: Fig.Generator = {
...
+            "Set the resulting package name, defaults to the directory name",
+          args: { name: "NAME" },
         },
       ],
     },
     {
       name: "init",
-      description: "Creates new cargo pkg in directory",
       icon: "📦",
+      description: "Creates new cargo pkg in directory",
```
Comment (Milo123459) [inline src/cargo.ts:284]:
```suggestion
      description: "Creates new cargo package in directory",
```

File: src/cargo.ts [hunk 3]
```diff
@@ -39,98 +39,356 @@ const binList: Fig.Generator = {
...
+          description: "Build all packages in workspace",
+        },
+        {
+          name: "--exclude",
+          description: "Exclude packages from the build",
+          args: { name: "SPEC" },
+        },
+        {
+          name: ["-j", "--jobs"],
+          description: "Number of parallel jobs, defaults to # of CPUs",
```
[the remark this answers is not in the document]
Comment (mschrage) [reply src/cargo.ts:120]: The original text seems closer to the `cargo` docs (which we should try to mirror when possible).

 https://doc.rust-lang.org/cargo/commands/cargo-build.html

[the remark this answers is not in the document]
Comment (Milo123459) [reply src/cargo.ts:120]: Personally I think this change makes it clearer but up to you

File: src/cargo.ts [hunk 4]
```diff
@@ -39,98 +39,356 @@ const binList: Fig.Generator = {
...
 const completionSpec: Fig.Spec = {
   name: "cargo",
   icon: "📦",
   description: "CLI Interface for Cargo",
   subcommands: [
     {
       name: ["build", "b"],
-      description: "Compile local package and dependencies",
       icon: "📦",
+      description: "Compile local package and dependencies",
```
[the remark this answers is not in the document]
Comment (grant0417) [reply src/cargo.ts:97]: I moved this down a line so the order of all the subcommands would be the same.

[the remark this answers is not in the document]
Comment (Milo123459) [reply src/cargo.ts:97]: Ah, that makes sense. I saw another change like this.

File: src/cargo.ts (hunk 2 above)
Comment (grant0417) [reply src/cargo.ts:284]: All of the text is directly from `cargo`

File: src/cargo.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (QuiiBz) [reply src/cargo.ts:120]: Let's keep the descriptions from cargo's doc.

File: src/cargo.ts [hunk 5]
```diff
@@ -39,98 +39,356 @@ const binList: Fig.Generator = {
...
+          description:
+            "Outputs a future incompatibility report at the end of the build (unstable)",
         },
       ],
     },
     {
-      name: ["run", "r"],
-      description: "Run a binary or example of the local package",
+      name: ["check", "c"],
+      icon: "📦",
```
[the remark this answers is not in the document]
Comment (QuiiBz) [reply src/cargo.ts:212]: Agree with @Milo123459, could you make the change @grant0417?

**Review by QuiiBz — APPROVED** [review]
This looks amazing, thank you so much for this PR!

**Pull request merged**